ZipBuilder/ZipBuilderV2Plan.md

```diff
@@ -0,0 +1,58 @@
+V2 will add an option to generate a zip distribution of binary frameworks from an arbitrary list
+of source and binary CocoaPods.
+
+## Introduction
+
+The current [Zip Builder](https://github.com/firebase/firebase-ios-sdk/tree/master/ZipBuilder)
+is Firebase specific. This is a proposal and initial plan
+to evolve the Zip Builder into a multi-purpose Apple binary framework creation tool.
+
+It would be useful to have a generic
+zip builder for other Google SDKs and other open source projects. In addition,
+providing a generic ZipBuilder would enable Firebase users to generate binary
+frameworks for configurations outside the standard Zip and Carthage distributions
+released in the standard Firebase release process.
+
+With the extra flexibility, the Zip Builder will be useful for both SDK distributors
+to package binary distributions and app developers who want a customized binary
+distribution that maps exactly to their app's requirements and provides a clean
+build time speed up.
+
+
+## Background
+
+A more flexible zip builder would enable the following scenarios:
+
+  * Creating a zip distribution from an arbitrary set of pods
+  * Support different building different sets of platform slices
+  * Support building with different Xcode versions
+  * Build only the subspecs needed for use case
+  * [Existing solutions](https://github.com/firebase/firebase-ios-sdk/issues/4284#issuecomment-552677044)
+  are intermittently maintained and written in Ruby. A Swift-implementation is
+  more accessible and maintainable by the Apple community.
+  * The Swift implementation will be easier to add Swift Package Manager support for which
+  there will likely be a need since Swift Package Manager is even more source-centric than
+  CocoaPods.
+
+## Plan
+
+  1. Add `--zipPods {JSON file}` option. The JSON file contains a list of CocoaPods
+  along with an optional version specifier. If the version is not specified, a CocoaPods
+  install will determine the version - typically the latest, unless another pod requires
+  something lower.
+
+  1. Add `--minimum-ios-version {version}` option. Specify the minimum iOS version to build for. Default
+  is 8.0.
+
+  1. Add `--archs {archs list}` option. Default is "arm64, arm64e, armv7, i386, x86_64"
```

Comment:
In the future when we add Catalyst support, how will we handle `x86_64`? Would it do both simulator and Catalyst, or would it rely on an extra flag being passed? 

I think default to simulator and needing an extra flag for Catalyst seems reasonable but thought I'd check.

Reply:
Hmm. We'll need to think more about how to evolve this towards adding Catalyst and eventually only building xcframeworks instead of frameworks when we can drop Xcode 10.

I'll clarify this in Future Directions.